examples_gbs/run_tutorial_similarity.py

```diff
@@ -0,0 +1,278 @@
+# calculating :math:`p_{k, n_{\max}}` really tough is the number of samples the corresponding
+# event contains! For example, the 6-photon event over 17 modes :math:`E_{k=6, n_{\max}=2}`
+# contains the following number of samples :
+
+print(similarity.event_cardinality(6, 2, 17))
+
+##############################################################################
+# To avoid calculating a large number of sample probabilities, an alternative for calculating the
+# event probability is to perform a Monte Carlo approximation. Here, samples within an event are
+# generated uniformly at random and their resultant probabilities are calculated. If :math:`N`
+# samples :math:`\{S_{1}, S_{2}, \ldots , S_{N}\}` are generated, then the event probability can
+# be approximated as
+#
+# .. math::
+#     p(E_{k, n_{\max}}) \approx \frac{1}{N}\sum_{i=1}^N p(S_i) |E_{k, n_{\max}}|,
+#
+# with :math:`|E_{k, n_{\max}}|` denoting the cardinality of the event.
+#
+# This method can be accessed using the :func:`~.prob_event_mc` function. The 4-photon event is
+# approximated as:
+
+print(similarity.prob_event_mc(nx.Graph(m0_a), 4, max_count_per_mode=2, n_mean=6))
+
+##############################################################################
+# The feature vector can then be calculated through Monte Carlo sampling using
+# :func:`~.feature_vector_mc`.
+#
+# .. note::
+#     The results of :func:`~.prob_event_mc` and :func:`~.feature_vector_mc` are probabilistic and
+#     may vary between runs. Increasing the optional ``samples`` parameter will increase accuracy
+#     but slow down calculation.
+#
+# The second method of MC approximation is intended for use in scenarios where it is
+# computationally intensive to pre-calculate a statistically significant dataset of samples from
+# GBS.
+#
+# Machine learning with GBS graph kernels
+# ---------------------------------------
+#
+# We have seen how GBS can be used to provide a mapping of graphs into a feature space. This
+# mapping can be used for machine learning tasks such as classification: by viewing each graph as
+# a point in the high dimensional space, we can use standard approaches from machine learning
+# such as `support vector machines <https://en.wikipedia.org/wiki/Support-vector_machine>`__ (SVMs).
+#
+# Let's build this up a bit more. The MUTAG dataset we are considering contains not only graphs
+# corresponding to the structure of chemical compounds, but also a *classification* of each
+# compound based upon its mutagenic effect. The four graphs we consider here have classifications:
+#
+# - MUTAG0: Class 1
+# - MUTAG1: Class 0
+# - MUTAG2: Class 0
+# - MUTAG3: Class 1
+
+classes = [1, 0, 0, 1]
+
+##############################################################################
+# Can we use GBS feature vectors to classify these graphs? We start by defining two-dimensional
+# feature vectors:
+
+events = [8, 10]

```

Comment:
Why 8 and 10? This choice may need a bit more motivation/explanation

Reply:
Have changed this by adding:

> The choice of ``events`` composing the feature vectors is arbitrary and we encourage the reader
to explore different combinations. Note, however, that odd photon-numbered events have zero
probability because ideal GBS only generates and outputs pairs of photons.

But appreciate suggestions!

Reply:
Maybe 'arbitrary' is not the right word? How about "there is freedom in the choice of events composing the feature vectors. We encourage the reader to explore different combinations

Reply:
Fixed!